### ISSUE TITLE
More a question: Can models be added to config with ease?

### ISSUE THREAD

**Issue by shirishr** [EpistasisLab/tpot#701]
[provide general introduction to the issue and why it is relevant to this repository]

I see that I can define my own config_dict and choose which classifier / regressor to keep.
Conversely can I add say lightgbm to my own config_dict and have TPOT do its magic?

**Comment by weixuanfu** [comment]
If the lightgbm has a scikit-learn API then the answer is yes.

**Comment by shirishr** [comment]
Looks like it does:

http://lightgbm.readthedocs.io/en/latest/Python-API.html#scikit-learn-api

I think I will try it out

**Comment by shirishr** [comment]
@weixuanfu,
Thank you for your encouragement.

I did built a custom_config with only **lightgbm** to my local config_dict. Tried IRIS.ipynb classification with it.

Seems to run well, but I have some newbie questions.

lightgbm.LGBMClassifier has a parameter **objective** which has possible values [‘binary’, ‘multiclass’ ] 

lightgbm.LGBMRegressor has parameter **objective** which has only one value [‘regression’ ] 

 lightgbm.LGBMRanker  has parameter **objective** which has only one value [‘lambdarank’]

lightgbm.LGBMClassifier needs a value for a parameter num_class which should be inherited from set(y_train) (obviously greater than 1) when **objective**  is **‘multiclass’** 

How do I manage these dependancies? I added num_class : [3] in the config for this test with IRIS dataset as a brute force method.

How can I default num_class from unique values of y_train?

I see that xgboost package also has a parameter called objective which for IRIS dataset would need a value  **'multi:softmax'**
